### DISCUSSION TITLE
Behaviour of `worker.on("error")` unclear

### DISCUSSION THREAD

**Discussion by Steffen911** [taskforcesh/bullmq#2939] [Q&A] [answered]
Hello all,

We're using bullmq to process events in our Node.js stack (Node 20.17, bullmq 5.12.10). During high-load phases we see errors like
```
Queue worker <queue-name> failed: Error: Client network socket disconnected before secure TLS connection was established
```
which is the message that we receive in the on error function like this:
```
    worker.on("error", (failedReason: Error) => {
      logger.error(
        `Queue job ${queueName} errored: ${failedReason}`,
        failedReason,
      );
      recordIncrement(convertQueueNameToMetricName(queueName) + ".error");
    });
```

Based on https://docs.bullmq.io/guide/workers it's not clear what happens to the given job at this point, especially since it seems to be an Redis connection issue (tracked in https://github.com/redis/ioredis/issues/1927). Does the job get marked as stalled as there are no update heartbeats and automatically retried?

**Comment by manast** [comment, marked as answer]
It will be marked as stalled and moved back to wait if more than the default 30 seconds have happened without the worker being able to extend the lock.

**Reply by manast** [reply]
It will by default only be moved to wait once by the way, but you can configure this to a larger number if this is something that happens often: https://api.docs.bullmq.io/interfaces/v5.WorkerOptions.html#maxStalledCount
Other wise the job will just fail.

**Reply by Steffen911** [reply]
Thank you 🙏
